readthedocs/builds/views.py

```diff
@@ -33,7 +34,24 @@ def get_queryset(self):
         return queryset
 
 
-class BuildList(BuildBase, ListView):
+class BuildTriggerMixin(object):
+    def post(self, request, project_slug):
+        project = get_object_or_404(
+            Project.objects.protected(self.request.user),
```

Comment:
These queryset methods are not documented at all, which we should probably fix, but `protected` is not the method we want here. This queryset will return projects with the `public` or `protected` privacy state, adding projects the user can view, from `Project.objects._add_user_repos`.

Instead, this should probably be `Project.objects.for_admin_user(self.request.user)`, as we only want to surface the form posting ability for users that are authenticated and own the project.

Reply:
Oh, I use the same than in `BuildBase`.

If I use `for_admin_user`, who will be able to trigger the builds? Just the owner? 

If that's the case, we will need to remove/hide the `Build` button if the user doesn't own the project, right?